test/junit/scala/collection/immutable/StreamTest.scala

```diff
@@ -15,4 +18,58 @@ class StreamTest {
     assertTrue(Stream(1,2,3,4,5).filter(_ < 4) == Seq(1,2,3))
     assertTrue(Stream(1,2,3,4,5).filterNot(_ > 4) == Seq(1,2,3,4))
   }
+
+  /** Test helper to verify that the given Stream operation allows
+    * GC of the head during processing of the tail.
+    */
+  def assertStreamOpAllowsGC[B](op: (=> Stream[Int], Int => B) => Any, f: Int => B): Unit = {
```

Comment:
It is making it possible, in a later commit, for me to retrofit test coverage of `.map` and `.flatMap` also allowing GC, in both the regular Stream and `.withFilter` varieties.